src/ssl.c

```diff
@@ -87,6 +87,9 @@
     #ifdef WOLFSSL_SHA512
         #include <wolfssl/wolfcrypt/sha512.h>
     #endif
+    #ifdef WOLFCRYPT_HAVE_SRP
```

Comment:
You are right. WOLFCRYPT_HAVE_SRP is nothing to do with OPENSSL_EXTRA.
I just thought only about wolfSSL_CTX_set_srp_password().
Because #include directive influence entire code, it should be placed outside OPENSSL_EXTRA.